include/onnxruntime/core/graph/graph.h

```diff
@@ -964,7 +964,7 @@ class Graph {
   */
   bool RemoveNode(NodeIndex node_index);
 
-  /** Add an edge between two Nodes.
+  /** Add an edge between two Nodes. The two node args at both ends of the edge must have the same type.
```

Comment:
>two node args [](http://example.com/codeflow?start=41&length=13)

are they allowed to be anything other than the same NodeArg?

Reply:
Yes. As long as they point to the same type.